src/app/system/manage-reports/create-report/create-report.component.ts

```diff
@@ -0,0 +1,124 @@
+
+  /** Report Form. */
+  reportForm: FormGroup;
+  /** Report Template Data. */
+  reportTemplateData: any;
+
+  /**
+   * Retrieves the report template data from `resolve`.
+   * @param {FormBuilder} formBuilder Form Builder.
+   * @param {SystemService} systemService Accounting Service.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {Router} router Router for navigation.
+   */
+  constructor(private formBuilder: FormBuilder,
+              private systemService: SystemService,
+              private route: ActivatedRoute,
+              private router: Router) {
+    this.route.data.subscribe((data: { reportTemplate: any }) => {
+      this.reportTemplateData = data.reportTemplate;
+    });
+  }
+
+  /**
+   * Creates the report form.
+   */
+  ngOnInit() {
+    this.createReportForm();
+    this.toogleVisibility();
+  }
+
+  /**
+   * Creates the report form.
+   */
+  createReportForm() {
+    this.reportForm = this.formBuilder.group({
+      'reportName': ['', Validators.required],
+      'reportCategory': [''],
+      'description': [''],
+      'reportType': ['', Validators.required],
+      'reportSubType': [{ value: '', disabled: true }],
+      'useReport': [false],
+      'reportSql': ['', Validators.required],
+      'reportParameters': this.formBuilder.array([])
+    });
+  }
+
+  /**
+   * Creates the report parameters form.
+   * @returns {FormGroup} Report Parameters Form.
+   */
+  createReportParametersForm(): FormGroup {
+    return this.formBuilder.group({
+      'id': [''],
+      'parameterId': ['', Validators.required],
+      'reportParameterName': ['']
+    });
+  }
+
+  /**
+   * Gets the Report Parameters Form.
+   * @returns {FormArray} Report Parameters Form.
+   */
+  get reportParameters(): FormArray {
+    return this.reportForm.get('reportParameters') as FormArray;
+  }
+
+  /**
+   * Adds a row in the report parameters form.
+   */
+  addReportParameter() {
+    this.reportParameters.push(this.createReportParametersForm());
+  }
+
+  /**
+   * Removes the report parameter row at given index from the report parameters form.
+   * @param {number} index Index of the row.
+   */
+  removeReportParameter(index: number) {
+    this.reportParameters.removeAt(index);
+  }
+
+  /**
+   * Toogles the visibility status of Report Sub Type dropdown.
+   */
+  toogleVisibility() {
+    this.reportForm.get('reportType').valueChanges
+      .subscribe(type => {
+        if (type === 'Chart') {
+          this.reportForm.get('reportSubType').enable();
+        } else {
+          this.reportForm.get('reportSubType').disable();
+        }
+      });
+  }
+
+  /**
+   * Submits the report form and creates report,
+   * if successful redirects to view created report.
+   */
+  submit() {
+    this.systemService.createReport(this.reportForm.value)
+      .subscribe((response: any) => {
+        this.router.navigate(['../view', response.resourceId], { relativeTo: this.route });
```

Comment:
But I am able to view report.

I guess you are trying to convey something else, please be more precise.

Reply:
While I was reviewing this PR i used demo server which has maker checker enabled for creating reports so when i create a report that report first goes to maker checker inbox then after checker's approval it available for use within the app where as in staging server maker checker is disabled that's why you are able to view report. 

Refer the email from @santoshconflux  explaining the same you are added in CC.

Screenshot of demo server Configuration Page.

![screencapture-demo-openmf-org-2019-05-12-18_40_45](https://user-images.githubusercontent.com/15383669/57582701-82a9a700-74e5-11e9-8019-4ecee334172b.png)

Screenshot of Demo server response when creating report with maker checker enabled.

![Screen Shot 2019-05-12 at 6 37 20 PM](https://user-images.githubusercontent.com/15383669/57582659-2d6d9580-74e5-11e9-9e41-fe5f5f9be220.png)

So since maker checker component is yet to be implemented in web-app leave a TODO comment here.